### PR TITLE
fix(weather): handle unknown and ambiguous cities safely

### LINKED ISSUE #19: [core] Improve weather city handling and unknown-city responses
## Overview

`core/weather.py` and `core/chat.py` have two related issues with how weather requests are handled.

**Broken case today:** `"Météo à Lyon ?"` → `detect_weather_city()` returns `None` (Lyon is not in `CITIES`) → `chat.py:112–113` returns `"Quelle ville ?"` even though the user clearly named a city.

There are also two overlapping but inconsistent keyword lists: `is_weather_query()` in `chat.py:40–42` and `detect_weather_city()` in `weather.py:43–52` use different sets of trigger words, which can cause one function to detect a weather query while the other misses it.

## What needs to be done

* [ ] Update `chat.py:112–113` to distinguish between two cases:
  - Weather query with a city name that is not in `CITIES` → `"Je ne connais pas encore cette ville."`
  - Weather query with no city mentioned at all → `"Quelle ville ?"`
* [ ] Align the keyword lists between `is_weather_query()` and `detect_weather_city()` so both functions use the same source of truth (a single constant defined in `config.py` or `weather.py`)
* [ ] Add a comment in `weather.py` documenting the current city coverage and its intentional scope

## Goal

Users asking about an unrecognized city receive an honest, clear message. `"Quelle ville ?"` is only shown when no city was mentioned. Keyword detection is consistent between the two functions.

**Priority: medium**

### PR DESCRIPTION
**Pull request by TheZupZup**
## Summary

- `detect_weather_city` now returns a discriminated result with four states instead of a plain tuple-or-None:
  - `(lat, lon, city)` — single recognized city → use the internal weather tool
  - `"multiple"` — two or more distinct supported cities detected → ask "Quelle ville ?"
  - `"no_city"` — weather keyword present but no city mentioned → ask "Quelle ville ?"
  - `"unknown_city"` — city-like word detected but not in the supported list → fixed refusal message
  - `None` — not a weather query → normal search/chat flow continues
- `chat()` dispatches all four states immediately and returns before reaching `should_search` or the LLM, so weather queries are fully contained.
- Removes `is_weather_query()`, which is now subsumed by `detect_weather_city`.
- Adds `tests/test_weather.py` covering all outcome branches, including dual-spelling deduplication (`montreal` / `montréal` treated as one city).

## Unsupported cities no longer hallucinate weather

Previously, a query like "météo à Paris" fell through to `is_weather_query`, which returned the generic "Quelle ville ?" — as if no city had been mentioned at all. Now it returns:

> Je n'ai pas accès à la météo pour cette ville.

No LLM is involved and no weather data is invented.

## Weather queries never fall back to web search or LLM guesses

All non-`None` returns from `detect_weather_city` are handled with an early `return` in `chat()`, before `should_search` or the normal chat path is reached. A weather query can no longer be routed to a web search or answered speculatively by the model.

Closes #19

---
_Generated by [Claude Code](https://claude.ai/code/session_0144tjDVZb2aRmdetoJ311kk)_

### CODE REVIEW

File: core/weather.py [hunk 1]
```diff
@@ -37,17 +38,68 @@ def get_weather(lat: float, lon: float, city: str) -> str:
...
+    "are", "me", "can", "you", "tell",
+    # single-letter fragments left after punctuation stripping
+    "s", "d", "l", "m", "n", "j", "t",
+}
+
+
+def _has_unrecognized_city(lower_input: str) -> bool:
+    """Returns True if the input contains a word that looks like an unsupported city name."""
+    words = re.sub(r"[^\w\s]", " ", lower_input).split()
+    return any(w not in _NOISE and len(w) > 1 for w in words)
```
[the remark this answers is not in the document]
Comment (TheZupZup) [reply core/weather.py]: Good catch — you're right.

Time/context words like "demain" or "maintenant" should not be treated as unknown cities. That causes valid "no city" queries to return the wrong message.

I'll adjust `_has_unrecognized_city` to ignore time-related words and only flag true unknown city tokens.

Thanks for spotting this 👍

**Comment by TheZupZup** [comment]
Improve weather city detection and handling

- Handle unknown and ambiguous cities safely
- Ignore time/context and weather descriptor words
- Improve parsing robustness for edge cases (numbers, hyphenated words)
- Add test coverage for router and memory parsing
- Fix lint issues (flake8)

Closes #19
